osu.Game/Input/TouchInputInterceptor.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Diagnostics;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.TypeExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Input.Events;
+using osu.Framework.Input.StateChanges;
+using osu.Framework.Logging;
+using osu.Game.Configuration;
+using osuTK;
+using osuTK.Input;
+
+namespace osu.Game.Input
+{
+    /// <summary>
+    /// Intercepts all positional input events and sets the appropriate <see cref="Static.TouchInputActive"/> value
+    /// for consumption by particular game screens.
+    /// </summary>
+    public partial class TouchInputInterceptor : Component
+    {
+        public override bool ReceivePositionalInputAt(Vector2 screenSpacePos) => true;
+
+        private readonly BindableBool touchInputActive = new BindableBool();
+
+        [BackgroundDependencyLoader]
+        private void load(SessionStatics statics)
+        {
+            statics.BindWith(Static.TouchInputActive, touchInputActive);
+        }
+
+        protected override bool Handle(UIEvent e)
+        {
+            bool touchInputWasActive = touchInputActive.Value;
+
+            switch (e)
+            {
+                case MouseEvent:
+                    if (e.CurrentState.Mouse.LastSource is not ISourcedFromTouch)
+                    {
+                        if (touchInputWasActive)
+                            Logger.Log($@"Touch input deactivated due to received {e.GetType().ReadableName()}", LoggingTarget.Input);
+                        touchInputActive.Value = false;
+                    }
+
+                    break;
+
+                case TouchEvent:
+                    if (!touchInputWasActive)
+                        Logger.Log($@"Touch input activated due to received {e.GetType().ReadableName()}", LoggingTarget.Input);
+                    touchInputActive.Value = true;
+                    break;
+
+                case KeyDownEvent keyDown:
+                    if (keyDown.Key == Key.T && keyDown.ControlPressed && keyDown.ShiftPressed)
+                        debugToggleTouchInputActive();
+                    break;
+            }
+
+            return false;
+        }
+
+        [Conditional("TOUCH_INPUT_DEBUG")]
+        private void debugToggleTouchInputActive()
+        {
+            Logger.Log($@"Debug-toggling touch input to {(touchInputActive.Value ? @"inactive" : @"active")}", LoggingTarget.Input, LogLevel.Debug);
```

Comment:
How would you feel about moving this to `LogLevel.Important` as to surface a message to the developer?

Reply:
If you mean a notification, then that would not be sufficient; I'd also have to change the `LoggingTarget` as `Input` log entries are suppressed from bubbling as notifications even if `Important`:

https://github.com/ppy/osu/blob/408ae841110854b17c963beb67c7aa68e9f55555/osu.Game/OsuGame.cs#L1155

I'm not opposed to it in principle, though.

Reply:
Eh, I guess it's fine as is. Kinda weird logic there.

Reply:
Sure, applied in 39ad91feea58e0a91e95b3a2be0889ff98be4ce6